utils/firestore.js

```diff
@@ -999,13 +1001,14 @@ const reportMissingSpecimen = async (siteAcronym, requestData) => {
 
 }
 
-const searchSpecimen = async (masterSpecimenId, siteCode) => {
+const searchSpecimen = async (masterSpecimenId, siteCode, allSitesFlag) => {
     const snapshot = await db.collection('biospecimen').where('820476880', '==', masterSpecimenId).get();
-    if(snapshot.size === 1) {
+    if (snapshot.size === 1) {
         const token = snapshot.docs[0].data().token;
         const response = await db.collection('participants').where('token', '==', token).get();
         const participantSiteCode = response.docs[0].data()['827220437'];
-        if(participantSiteCode === siteCode) return snapshot.docs[0].data();
+        if (allSitesFlag) return snapshot.docs[0].data();
+        else if (!allSitesFlag && participantSiteCode === siteCode) return snapshot.docs[0].data();
```

Comment:
reason I added !allSitesFlag is allSitesFlag variable could be set to undefined & thats a very edge case. To prevent that, I had to add  !allSitesFlag safeguard.